src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/prototype/TempMultipleAlignmentsReclassifier.java

```diff
@@ -0,0 +1,211 @@
+                                                                                       final int uniqReadLenInclusive) {
+        Utils.validateArg(originalConfiguration.size() > 2,
+                "assumption that input configuration to be fine tuned has more than 2 alignments is violated.\n" +
+                        originalConfiguration.stream().map(AlignmentInterval::toPackedString).collect(Collectors.toList()));
+
+        // two pass, each focusing on removing the alignments of a contig that offers low uniqueness in one sense:
+
+        // first pass is for removing alignments with low REFERENCE UNIQUENESS, using low mapping quality as the criterion
+        final List<AlignmentInterval> selectedAlignments = new ArrayList<>(originalConfiguration.size()),
+                                      lowUniquenessMappings = new ArrayList<>(originalConfiguration.size());
+
+        for (final AlignmentInterval alignment : originalConfiguration) {
+            if (alignment.mapQual >= mapQThresholdInclusive)
+                selectedAlignments.add(alignment);
+            else
+                lowUniquenessMappings.add(alignment);
+        }
+
+        // second pass, the slower one, is to remove alignments offering low READ UNIQUENESS,
+        // i.e. with only a very short part of the read being uniquely explained by this particular alignment;
+        // the steps are:
+        //      search bi-directionally until cannot find overlap any more, subtract from it all overlaps.
+        //      This gives unique read region it explains.
+        //      If this unique read region is "short": shorter than {@code uniqReadLenInclusive}), drop it.
+
+        // each alignment has an entry of a tuple2, one for max overlap front, one for max overlap rear,
+        // max overlap front is a tuple2 registering the index and overlap bases count
+        final List<Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>> maxOverlapMap =
+                getMaxOverlapPairs(selectedAlignments);
+
+        final List<Integer> idxToRemove = new ArrayList<>(selectedAlignments.size());
+        for (int i = 0; i < selectedAlignments.size(); ++i) {
+            final AlignmentInterval cur = selectedAlignments.get(i);
+            final Tuple2<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> maxOverlapFrontAndRear = maxOverlapMap.get(i);
+            final int maxOverlapFront = Math.max(0, maxOverlapFrontAndRear._1._2);
+            final int maxOverlapRead = Math.max(0, maxOverlapFrontAndRear._2._2);
+
+            final int uniqReadSpan = cur.endInAssembledContig - cur.startInAssembledContig + 1 - maxOverlapFront - maxOverlapRead;
+            if (uniqReadSpan < uniqReadLenInclusive)
+                idxToRemove.add(i);
+        }
+
+        if ( idxToRemove.isEmpty() )
+            return new Tuple2<>(selectedAlignments, lowUniquenessMappings);
+
+        // removing in reverse order so that iterators are not invalidated if we were to remove from start
```

Comment:
It's ok to remove via iterator if traversing forward. I think you could do this all in one loop instead of re-iterating over the list. Just iterate over selected alignments, test the overlap from the, and remove it via the iterator if the overlap is too high.

Reply:
done